wasm_validator/derive/src/lib.rs

```diff
@@ -0,0 +1,228 @@
+//! Crate with validator-related derive macros.
+
+#![allow(clippy::panic)]
+
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{parse_macro_input, parse_quote, DeriveInput};
+
+mod conversion;
+mod entrypoint;
+mod token;
+mod validate;
+
+/// Annotate the user-defined function that starts the execution of a validator.
+///
+/// Validators are only checking if an operation is **invalid**, not if it is valid.
+/// A validator can either deny the operation or pass it to the next validator if there is one.
+///
+/// # Attributes
+///
+/// This macro must have an attribute describing entrypoint parameters.
+///
+/// The syntax is:
+/// `#[iroha_wasm::validator_entrypoint(params = "[<type>,*]")]`, where `<type>` is one of:
+/// - `authority` is a signer account id who submits an operation
+/// - `transaction` is a transaction that is being validated
+/// - `instruction` is an instruction that is being validated
+/// - `query` is a query that is being validated
+/// - `expression` is an expression that is being validated
+///
+/// Exactly one parameter of *operation to validate* kind must be specified.
+/// `authority` is optional.
+/// Parameters will be passed to the entrypoint function in the order they are specified.
+///
+/// ## Authority
+///
+/// A real function parameter type corresponding to the `authority` should have
+/// `iroha_wasm::data_model::prelude::AccountId` type.
```

Comment:
```suggestion
/// [`iroha_wasm::data_model::prelude::AccountId`] type.
```

I think here and further it makes sense to leverage [Rustdoc linking feature](https://doc.rust-lang.org/rustdoc/write-documentation/linking-to-items-by-name.html), so it will be able to catch outdated/wrong links if they occur.

Reply:
I'd like to, but I can't. `iroha_wasm_validator_derive` does not depend on `iroha_wasm` or `iroha_wasm_validator`